bionic/core/flow_execution.py

```diff
@@ -155,18 +162,70 @@ def _process_active_entry(self, entry):
             and not entry.state.task.is_simple_lookup
         )
         if can_compute_remotely:
-            new_state_for_subprocess = entry.state.strip_state_for_subprocess()
-            # We want serial execution in the subprocesses.
+            # When we run an entry remotely, we also need to run all of its immediate
+            # non-persistable ancestors, since their values can't be shared between
+            # processes. Those ancestors may have follow-up tasks, so we'll need to
+            # run those too; and if those follow-ups are persistable (but not actually
+            # persisted yet), their values will be usable everywhere, so we'll want to
+            # track the fact that they're running. So really we're going to run a
+            # collection of persistable entries (including the original entry), plus
+            # their immediate non-persistable ancestors.
+            remote_subgraph = RemoteSubgraph(entry.state, self._bootstrap)
+            persistable_subgraph_entries = [
+                self._get_or_create_entry_for_state(persistable_subgraph_state)
+                for persistable_subgraph_state in remote_subgraph.persistable_but_not_persisted_states
+            ]
+            assert entry in persistable_subgraph_entries
+
+            # We want all of our entries to be initialized before we try running
+            # them remotely. We'll just require this of the persistable ones, since
+            # all the other ones are ancestors of them.
+            for persistable_subgraph_entry in persistable_subgraph_entries:
+                if persistable_subgraph_entry == entry:
+                    assert persistable_subgraph_entry.level >= EntryLevel.INITIALIZED
+                else:
+                    self._add_requirement(
+                        src_entry=entry,
+                        dst_entry=persistable_subgraph_entry,
+                        level=EntryLevel.INITIALIZED,
+                    )
+            if not entry.all_outgoing_reqs_are_met:
+                self._mark_entry_blocked(entry)
+                return
+
+            target_entries = [
+                persistable_subgraph_entry
+                for persistable_subgraph_entry in persistable_subgraph_entries
+                if persistable_subgraph_entry.stage
+                not in (EntryStage.COMPLETED, EntryStage.IN_PROGRESS)
+            ]
+            for target_entry in target_entries:
+                if target_entry.stage == EntryStage.PENDING:
+                    self._mark_entry_active(target_entry)
```

Comment:
This means that more than one entry can be active now. We should change the documentation of `EntryStage.ACTIVE` to reflect this.